syntax = "proto3";

package tensorflow.serving;
option cc_enable_arenas = true;

import "tensorflow/core/framework/tensor.proto";
import "tensorflow_serving/apis/model.proto";

message FeatureScore {
  repeated uint64 id = 1;
  repeated float score = 2;
}

message FeatureScoreVector {
  repeated FeatureScore feature_score = 1;
}

// PredictRequest specifies which XGBoostFM model to run, as well as
// the XGBoost inputs and the FM inputs.
message PredictRequest {
  // Model Specification. If version is not specified, will use the latest
  // (numerical) version.
  ModelSpec model_spec = 1;

  // Input FeatureScoreVector.
  // One for the XGBoost inputs and the other for the FM inputs.
  map<string, FeatureScoreVector> inputs = 2;
}

// Response for PredictRequest on successful run.
message PredictResponse {
  // Effective Model Specification used to process PredictRequest.
  ModelSpec model_spec = 2;

  // Output tensors.
  map<string, TensorProto> outputs = 1;
}
